test/whitespace/functions.unit

```diff
@@ -165,4 +165,12 @@ main() {
 >>> generic function typed parameter
 function(int   foo  <  T  ,S >(T t, S s)) {}
 <<<
-function(int foo<T, S>(T t, S s)) {}
\ No newline at end of file
+function(int foo<T, S>(T t, S s)) {}
+>>> decorator-style function
+function() => decorator(() {
+  body;
+});
+<<<
+function() => decorator(() {
+  body;
+});
```

Comment:
What does the output look like if:

* The parameter list for the function has to wrap.
* The argument list for the decorator call before the function has to wrap.
* The argument list for the decorator call after the function has to wrap.